### PR TITLE
Infrastructure to remove config options if incompatible with current version

### LINKED ISSUE #8: manage --with-feature compatibility with versions
With 11 coming, --with-llvm will be available. But it's only avaialble starting on this version. Maybe it's time we add an array of flags with compatible versions (and we use it to cleanup the CONFIGOPTS on the fly ?

### PR DESCRIPTION
**Pull request by marco44**
Close #8. But cannot be merged before #7, as it depends on it :)

### CODE REVIEW

**Comment by rjuju** [comment]
Then it's a good thing I already merged #7 :)

Should we also tag the `--with-llvm` with a 11.0 too so it'll be forward compatible when it'll be released?  Also, I'm not sure of how it should be done later, always add `dev` and `$MAJOR_VERSION` entry if the hash for each new flag?

**Comment by rjuju** [comment]
Should we also care about removed flags?

**Comment by marco44** [comment]
I've added all that has been asked. There was not a lot I could do to test it for now though :)

File: postgres.pl [hunk 1]
```diff
@@ -369,6 +378,39 @@ sub setenv
...
+                                $config=~ s/$param//;
+                        }
+                }
+        }
+        foreach my $paramversion(keys (%deprecated_configopts_per_version))
+        {
+                if (compare_versions($version,$paramversion)==1)
+                {
+                        # This version is newer than paramversion, so we remove these options
+                        my @to_remove=@{$new_configopts_per_version{$paramversion}};
```
Comment (rjuju) [inline postgres.pl]: Copy/paste error here :)

File: postgres.pl [hunk 2]
```diff
@@ -88,6 +88,15 @@
...
 );
 
+# New configopts per version
+my %new_configopts_per_version=(
+        '11' => ['--with-llvm'],
+        'dev' => ['--with-llvm']);
+
+# No idea what version it could be, but lets's do this right now
+my %deprecated_configopts_per_version=(
+        );
```
Comment (rjuju) [inline postgres.pl]: `--with-wal-segsize` has been removed in 11/dev, we could add it to test this one

**Review by rjuju — DISMISSED** [review]

**Review by rjuju — APPROVED** [review]

**Pull request merged**